src/linter.rs

```diff
@@ -325,7 +362,7 @@ pub fn lint_fix(
         );
 
         // Apply autofix.
-        if let Some((fixed_contents, applied)) = fix_file(&diagnostics, &locator) {
+        if let Some((fixed_contents, applied)) = fix_file(&result.data, &locator) {
```

Comment:
Having access to the parse error here means that we can also catch errors that _we_ introduced.